### ISSUE TITLE
OAuth not working with latest version of 24.bot-authentication-msgraph

### ISSUE THREAD

**Issue by Anshul21** [microsoft/BotBuilder-Samples#1655]
## Sample information
1. Sample type: 24.bot-authentication-msgraph.sln
2. Sample language: dotnetcore
3. Sample name: 24.bot-authentication-msgraph

## Describe the bug
After upgrading to latest version of  24.bot-authentication-msgraph. The OAuth is not working fine in emulator as well as teams. 

## To Reproduce
1. Go to  24.bot-authentication-msgraph
2. Enter your connectionName, bot credentials
3. Run the bot locally.
4. use the bot in emulator to reproduce the error.
## Expected behavior
Present auth card to user to log in 
## Screenshots

<img width="616" alt="Screen Shot 2019-07-16 at 4 03 16 PM" src="https://user-images.githubusercontent.com/7298709/61336386-fddd5100-a7e5-11e9-8251-e0c8061211a3.png">
<img width="664" alt="Screen Shot 2019-07-16 at 4 03 27 PM" src="https://user-images.githubusercontent.com/7298709/61336389-ffa71480-a7e5-11e9-902f-4000c95567d2.png">

**Comment by dcmesh** [comment]
I'm getting a pretty similar error for my bot that uses an OAuth connection to Microsoft Graph as well. 
![image](https://user-images.githubusercontent.com/36556579/61459250-54f43a80-a93a-11e9-88ba-cab8627d43dc.png).

Side Note: another bot that I have which uses OAuth sign-in has the Sign In prompt appear, but nothing happens with it.

**Comment by Anshul21** [comment]
@dcmesh hey! Downgrade your emulator to v4.4.0

**Comment by dcmesh** [comment]
@Anshul21 Thank you for the reply! I just tried that, but it did not end up working.

**Comment by anusharavindrar** [comment]
Could you post on what behavior you observe when you click on 'Confirm'? Also, what behavior are you encountering when you use Teams?

**Comment by anusharavindrar** [comment]
@Anshul21 Could you post on what behavior you observe when you click on 'Confirm'? Also, what behavior are you encountering when you use Teams?

**Comment by Anshul21** [comment]
@dcmesh Also change your emulator settings. I hope this works.
![image](https://user-images.githubusercontent.com/7298709/61478213-ec12bf80-a945-11e9-8575-1204104112a1.png)

**Comment by Anshul21** [comment]
> @Anshul21 Could you post on what behavior you observe when you click on 'Confirm'? Also, what behavior are you encountering when you use Teams?

On Confirm, It doesn't redirect me to the login page and it just comes back with token = "whatever you see after &&& in the url"

**Comment by danwkennedy** [comment]
I'm seeing this as well. I've also noticed that 4.4.x versions of the emulator won't connect to my bot running on localhost so I can't downgrade either. 

Whenever I click the confirm link in the emulator the token returned to the dialog is of the format:
`<ConversationId>|livechat`

**Comment by sschutten** [comment]
I'm experiencing the same thing as mentioned by @danwkennedy in 4.5.2. I've also tried 4.4.0 and 4.4.2 but those versions don't trigger the OnTokenResponseEventAsync after a successful sign in. Only if I enable the sign-in verification code for OAuthCards it works in 4.4.x.

**Comment by dcmesh** [comment]
@Anshul21 If your still having issues, these fixes were able to help me:

- Try running the bot emulator in Administrator Mode
- Make sure that the AAD App Registration you are connecting to is properly listed in the appsettings.json file, with something like `"ConnectionName": "<insert_connection_name>",`
- Make sure that the AAD App Registration has the proper Redirect URI: `https://token.botframework.com/.auth/web/redirect`
- Make sure that the OAuth Connection Setting has its connection to the AAD App Registration configured to the correct version of OAuth (v1 or v2), and that it is the same in the Bot Emulator's settings

**Comment by Dolphin02** [comment]
> I'm getting a pretty similar error for my bot that uses an OAuth connection to Microsoft Graph as well.
> ![image](https://user-images.githubusercontent.com/36556579/61459250-54f43a80-a93a-11e9-88ba-cab8627d43dc.png).
> 
> Side Note: another bot that I have which uses OAuth sign-in has the Sign In prompt appear, but nothing happens with it.

Hi, I'm getting a pretty similar error for my bot that uses an OAuth connection to Microsoft Graph as well. Have you figured out it?

**Comment by dcmesh** [comment]
@Dolphin02 Another issue I ran into was trying to run the bot behind a proxy. I think some part of the OAuth connection can be interfered with by a proxy, even if the Bot Framework Emulator is set to override to localhost when bypassing ngrok. If you are trying to authenticate with your bot behind behind a proxy, that may be the source of the issue.

**Comment by sgellock** [comment]
@carlosscastro can you take a look please.

**Comment by textminer** [comment]
Hi, I'm having same issue using Bot Emulator 4.5.2

I made sure to:
- Run Bot Emulator as an admin.
- Bot Emulator settings with: Use a sign-in verification code for OAuthCards and Bypass ngrok for local addresses. (also without them)
- Created App registration ini AAD and Redirect is type Web - https://token.botframework.com/.auth/web/redirect
- Added Connection String in Azure settings, and tested connection successfully.
- Connection string is available and passed correctly to OAuthPrompt() dialog in connectionName property.
- Verified that Bot.exe process has all accesses in windows firewall
- I'm not running behind a proxy.

After I press confirm in popup as seen in picture, system hangs. Also checked using Fiddler and no Oauth request goes to AAD. Even sending a message to the bot does not progress.

Please, advice since OauthCards is the only way to solve my architecture.

![imagen](https://user-images.githubusercontent.com/38899601/62874339-6e00c900-bcf7-11e9-8d66-d902ab3043fe.png)

**Comment by sindhugururaj** [comment]
Am having the same issue too. Could anyone help with a workaround? 

thank you

**Comment by Anshul21** [comment]
@sindhugururaj @textminer

Do the following 
1. Downgrade the bot emulator to 4.4.0
2. Go to emulator settings, in check automatic updates, check "use v1 authentication" and also check the "use cards for oauth". 
3. You should be able to run the bot

**Comment by sindhugururaj** [comment]
Tried the above steps.

![image](https://user-images.githubusercontent.com/27359060/62876696-8969d500-bcf3-11e9-88c0-909f97399af4.png)

am getting a pop up saying bad request. my .env file contains connectionName, microsoft AppID and app password
the outhPrompt 
 this.addDialog(new OAuthPrompt(OAUTH_PROMPT, {
            connectionName: process.env.connectionName,
            text: 'Please Sign In',
            title: 'Sign In',
            timeout: 300000
        }));


added these settings to the bot emulator. 
![image](https://user-images.githubusercontent.com/27359060/62876845-d3eb5180-bcf3-11e9-9132-61c5515bfa98.png)


is there anything that am missing?

**Comment by textminer** [comment]
With Bot Framework 4.4.0, it does not even open the popup.

**Comment by sgellock** [comment]
@carlosscastro can you jump on this please?  thanks

**Comment by carlosscastro** [comment]
Looking into this :)

**Comment by carlosscastro** [comment]
Hi all - I tracked this down to be an emulator bug due to recent changes. It is being tracked here: https://github.com/microsoft/BotFramework-Emulator/issues/1728. 

I got to reproduce the same issue. On emulator version 4.2, my OAuthPRompt worked. Then upgraded to the latest and saw the error reported in the bug. 

However, I was able to find out why it happens - ngrok is misconfigured in the emulator. Ngrok path in emulator settings (lower bottom right) could be set to nothing or, as it was in my case, set to a path that did not actually have ngrok.exe. 

When I corrected the path to ngrok.exe in emulator settings, it started working, so that should fix it for you.

Once the emulator bug (https://github.com/microsoft/BotFramework-Emulator/issues/1728) is resolved, we will show a clear error when ngrok is misconfigured, which will make for a much better user experience.

**Comment by carlosscastro** [comment]
Closing the issue since the proposed solution should unblock you. If you don't get unblocked by this, please reopen. If there is another, clearly different issue, feel free to open a separate bug. 
Thanks!

**Comment by sgellock** [comment]
@carlosscastro thanks

**Comment by textminer** [comment]
Hi @carlosscastro,

With emulator version 4.2 I can't even get to the OauthCards prompt. It does not show up and hangs the bot. While as with the version 4.5.2 I can get to the OauthCards prompt and hangs after clicking the Login button. I also made sure that the path to ngrok (v2.3.34) is ok. 
As seen in the image the ngrok.exe process starts with the Emulator and I whitelisted it in the Windows Firewall. Verified that changing user settings Use a sign... and Use version 1.0 tokens, and still reproduces.
So, you make it work on version 4.5.2, with ngrok configured and which user settings?

![imagen](https://user-images.githubusercontent.com/38899601/63118762-eaeca680-bf74-11e9-8729-429aa7dc3761.png)

**Comment by sathishreddy48** [comment]
Could you please provide connectionstring : sample

**Comment by sathishreddy48** [comment]
Authentication is not working in teams

**Comment by victorjoel44** [comment]
did you fix it?
